hm2task2.c

```diff
@@ -0,0 +1,53 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+
+float expLinear(float number, int power) 
+{
+    if (power == 0) 
+        return 1;
+    float result = 1;
+    int pow = abs(power);
+    for (int i = 1; i <= pow; i++)
+    {
+        result *= number;
+    }
+    if (power < 0) return (1 / result);
+    return result;
+}
+
+float expLogN(float number, int power)
+{
+    if (power == 0) 
+        return 1;
+    int pow = abs(power);
+    float result = 1;
+    while (pow != 0) 
+    {
+        if (pow & 1)
+        {
+            result *= number;
+            pow--;
+        }
+        number *= number;
+        pow = (int)pow / 2;

```

Comment:
```suggestion
        pow = pow / 2;
```
Приведение инта к инту?